modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -16,33 +17,42 @@ def create_a_record(fqdn, ip)
       if found = dns_find(fqdn)
         raise(Proxy::Dns::Collision, "#{fqdn} is already used by #{found}") if found != ip
       else
-        zone = match_zone(fqdn)
+        zone = match_zone(fqdn, enum_zones)
         msg = "Added DNS entry #{fqdn} => #{ip}"
         cmd = "/RecordAdd #{zone} #{fqdn}. A #{ip}"
         execute(cmd, msg)
       end
     end
 
-    # noop
     def create_ptr_record(fqdn, ip)
-      found = dns_find(ip)
-      raise(Proxy::Dns::Collision, "#{ip} is already used by #{found}") if found && found != fqdn
-      true
+      if found = dns_find(ip)
+       raise(Proxy::Dns::Collision, "#{ip} is already used by #{found}") if found != fqdn
+      else
+        ptr_record = get_ptr_record(ip)
+        zone = match_zone(ptr_record, enum_zones)
+        msg = "Added PTR entry #{ip} => #{fqdn}"
+        cmd = "/RecordAdd #{zone} #{ptr_record}. PTR #{fqdn}."
+        execute(cmd, msg)
+      end
     end
 
     def remove_a_record(fqdn)
       ip = dns_find(fqdn)
       raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{fqdn}") unless ip
-      zone = match_zone(fqdn)
+      zone = match_zone(fqdn, enum_zones)
       msg = "Removed DNS entry #{fqdn} => #{ip}"
       cmd = "/RecordDelete #{zone} #{fqdn}. A /f"
       execute(cmd, msg)
     end
 
-    # noop
     def remove_ptr_record(ip)
-      raise Proxy::Dns::NotFound.new("Cannot find DNS entry for #{ip}") unless dns_find(ip)
-      true
+      fqdn = dns_find(ip)
```

Comment:
>Spent the whole afternoon testing smart-proxy against it.

Lucky you!

If you allow me one off topic question: Did smart-proxy successfully remove DHCP reservations? This is really bugging me since the time I started out with foremen.